### PR TITLE
Support GitHub token for pushing gh-pages

### PR DESCRIPTION
**Pull request by jirikuncar**
Provide an alternative authentication method for pushing to gh-pages.

### CODE REVIEW

**Comment by jirikuncar** [comment]
@Carreau thanks! I have fixed a typo in docs.

File: chartpress.py [hunk 1]
```diff
@@ -15,11 +15,26 @@
 
 __version__ = '0.2.0.dev'
 
+# name of the environment variable with GitHub token
+GITHUB_TOKEN_KEY = 'GITHUB_TOKEN'
```
Comment (minrk) [inline chartpress.py]: You've defined this variable, but haven't used it below. Since it's used only once, I think we can drop the variable.

Comment (jirikuncar) [reply chartpress.py]: Thanks!

**Comment by jirikuncar** [comment]
@minrk is there anything else I should fix?

**Comment by minrk** [comment]
Thanks!

**Pull request merged**